src/Gateway/PayplugCreditCard.php

```diff
@@ -248,4 +247,111 @@ public function payment_fields()
 			$this->saved_payment_methods();
 		}
 	}
+
+	/**
+	 * Process the subscription scheduled payment
+	 */
+	public function scheduled_subscription_payment($amount, $order) {
+
+		$order_id      = PayplugWoocommerceHelper::is_pre_30() ? $order->id : $order->get_id();
+		$customer_id = PayplugWoocommerceHelper::is_pre_30() ? $order->customer_user : $order->get_customer_id();
+
+		$payplug_meta = wcs_get_subscription($order->get_meta('_subscription_renewal'))->get_parent()->get_meta("_payplug_metadata");
```

Comment:
this variable should be named `$subscription`

Reply:
that specifc variable isn't the subscription, is the payment metadata, next commit I splitted it to be more clear.